guardian/nla-pageview-job
# Update Node to v14 and bump some old dependencies too

## What does this change?
Upgrade to Node 14 (that's the latest supported by Lambda according to [this page](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html))

While we're here we're also bumping some dependencies as there are a few known issues with e.g. `lodash` and `bl` which were implicated by `archiver` and `async`.

## How to test
If it builds, update the cloudformation for our CODE version and deploy it. We should be able to trigger the CODE lambda manually or with a scheduled event. Assuming it runs successfully we should see some generated log entries and output file(s) in S3.
**Update:** Tested and verified that the new code is running in CODE, files were created but the FTP upload process, although triggered, timed out while attempting to connect to the remote server. This may be expected in CODE.


## How can we measure success?
If it works, success!

## Have we considered potential risks?
If it breaks we'll have to revert the cloudformation and redeploy the previous known good build while we rethink.

## Images
N/A

## Accessibility
N/A
